src/coreComponents/mesh/CellElementRegion.cpp

```diff
@@ -35,15 +72,157 @@ CellElementRegion::CellElementRegion( string const & name, Group * const parent
 CellElementRegion::~CellElementRegion()
 {}
 
-void CellElementRegion::generateMesh( Group const & cellBlocks )
+
+void CellElementRegion::registerSubRegion( CellBlockABC const & cellBlock )
 {
   Group & elementSubRegions = this->getGroup( viewKeyStruct::elementSubRegions() );
+  // For now, subRegion name must be the same as the cellBlock (so we can match them and reference them in errors).
+  CellElementSubRegion & subRegion =
+    elementSubRegions.registerGroup< CellElementSubRegion >( cellBlock.getName() );
+
+  subRegion.copyFromCellBlock( cellBlock );
+}
+
+string CellElementRegion::getCellBlockAttributeValue( string_view cellBlockName )
+{
+  size_t const sep = cellBlockName.find( '_' );
+  return sep != string::npos ? string( cellBlockName.substr( 0, sep ) ) : string( "" );
```

Comment:
Can't you use the `stringutils::tokenize` the `cellBblockName` and return the first one if `size > 1` ?

Reply:
I found `stringutilities::removeStringAndFollowingContent()` that seems perfect :)

Reply:
Great, I did not know about it 🤣